### PR TITLE
Added tests for main

### PR DESCRIPTION
**Pull request by carlosvdr**

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+class TestScript:
+    @pytest.fixture
+    def pgsql(self, postgresql):
+        return postgresql
+
+    async def test_allocated_subgraph_watcher(self, pgsql):
+        subgraph1 = "QmPnu3R7Fm4RmBF21aCYUohDmWbKd3VMXo64ACiRtwUQrn"
+        subgraph2 = "QmTJBvvpknMow6n4YU8R9Swna6N8mHK8N2WufetysBiyuL"
+        with mock.patch("autoagora.main.init_config") as mock_init_config:
+            # with mock.patch("autoagora.main.asyncpg") as mock_asyncpg:
```
Comment (aasseman) [inline tests/test_main.py:27]:
```suggestion
```

File: tests/test_main.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                "autoagora.main.get_allocated_subgraphs"
+            ) as mock_get_allocated_subgraphs:
+                with mock.patch("autoagora.main.set_cost_model") as mock_set_cost_model:
+                    with mock.patch(
+                        "autoagora.main.model_update_loop"
+                    ) as mock_model_update_loop:
+                        with mock.patch(
+                            "autoagora.main.price_bandit_loop"
+                        ) as mock_price_bandit_loop:
+                            mock_init_config.return_value = None
```
Comment (aasseman) [inline tests/test_main.py:38]: Doesn't it already always return `None`?

**Review by aasseman — CHANGES_REQUESTED** [review]

File: tests/test_main.py (hunk 2 above)
Comment (carlosvdr) [reply tests/test_main.py:38]: oh yeah, you are right that was from some test  was doing, this one needs some clean up

File: tests/test_main.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    metrics_server,
+)
+
+
+class TestScript:
+    @pytest.fixture
+    def pgsql(self, postgresql):
+        return postgresql
+
+    async def test_allocated_subgraph_watcher(self, pgsql):
```
[the remark this answers is not in the document]
Comment (carlosvdr) [reply tests/test_main.py:23]: Yeah just tried it, I did for another one but wasnt able to perhaps due to the way the code is done.

**Review by aasseman — APPROVED** [review]
Please rebase onto master, squash, and merge

**Pull request merged**